### PR TITLE
[Feature][st-engine] add server in NodeExtension

### LINKED ISSUE #2210: [STIP-3] [Server design][st-engine] st new engine server design
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

As shown below

Use Hazelcast's NodeExtension to extend the implementation of the engine, relying on Hazelcast dynamic scaling, automatic master selection, and distributed memory.
The gray part in the figure is hezalcast, and the blue part is ST-engine.

job submit：
The client submits a job to the ST-Service of the Master node, Job generates a physical execution plan Task in ST-Service and submits it to ST-Service of other nodes in the cluster for execution.
Information such as its job status is maintained in Hazelcast's Imap (distributed storage).

fault tolerance:
The ST-Service instance is stateless (the state is stored in Hazelcast's distributed storage), so when the Master node hangs up. Hazelcast will automatically elect another node as the Master, and the ST-Service on this node will continue to provide services.

image

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ic4y**
## Purpose of this pull request
this closed #2210 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Review by CalvinKirs — CHANGES_REQUESTED** [review]
hi, you need follow New license user guide

**Comment by CalvinKirs** [comment]
could you check again? you add the new jar, but the License file only adds one column.
 btw, why did we don't use 5.x?

has any issue or design with this PR?

**Comment by ic4y** [comment]
@CalvinKirs 
5.X Hazelcast and JET are under the same module, we don't want to introduce JET.
and checked by checkLicense.sh, it is true that only hazelcast-4.2.jar

**Comment by CalvinKirs** [comment]
> @CalvinKirs 5.X Hazelcast and JET are under the same module, we don't want to introduce JET. and checked by checkLicense.sh, it is true that only hazelcast-4.2.jar

I don't think this is a good reason, often newer versions get more maintenance, bug fixes, etc. I think we should discuss this. 

and any design about this? You submitted a large module without any description.

File: tools/dependencies/known-dependencies.txt [hunk 1]
```diff
@@ -731,4 +731,5 @@ zookeeper-jute-3.5.9.jar
 zstd-jni-1.3.3-1.jar
 zstd-jni-1.4.3-1.jar
 jakarta.activation-api-1.2.1.jar
-jakarta.xml.bind-api-2.3.2.jar
\ No newline at end of file
+jakarta.xml.bind-api-2.3.2.jar
+hazelcast-4.2.jar
```
Comment (CalvinKirs) [inline tools/dependencies/known-dependencies.txt]: So please don't make extraneous formatting modifications. even if it's small.

Comment (ic4y) [reply tools/dependencies/known-dependencies.txt]: This is caused by the `\n`. Adding a new line requires a `\n`.

**Comment by ic4y** [comment]
> > @CalvinKirs 5.X Hazelcast and JET are under the same module, we don't want to introduce JET. and checked by checkLicense.sh, it is true that only hazelcast-4.2.jar
> 
> I don't think this is a good reason, often newer versions get more maintenance, bug fixes, etc. I think we should discuss this.
> 
> and any design about this? You submitted a large module without any description.

Ok i upgraded to 5.1，This related design is in #2210

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**